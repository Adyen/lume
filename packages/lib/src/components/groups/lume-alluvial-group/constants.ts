export const NODE_LABEL_PADDING = 8;
export const GHOST_STROKE_WIDTH_OFFSET = 8;
