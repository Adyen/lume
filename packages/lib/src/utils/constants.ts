export enum Colors {
  Skyblue = 'skyblue',
  Royalblue = 'royalblue',
  Violet = 'violet',
  Darkteal = 'darkteal',
  Gold = 'gold',
}

export enum SequentialColors {
  Skyblue10 = 'skyblue-10',
  Skyblue20 = 'skyblue-20',
  Skyblue30 = 'skyblue-30',
  Skyblue40 = 'skyblue-40',
  Skyblue50 = 'skyblue-50',
  Skyblue60 = 'skyblue-60',
  Skyblue70 = 'skyblue-70',
  Skyblue80 = 'skyblue-80',
  Skyblue90 = 'skyblue-90',
  Skyblue100 = 'skyblue-100',

  Royalblue10 = 'royalblue-10',
  Royalblue20 = 'royalblue-20',
  Royalblue30 = 'royalblue-30',
  Royalblue40 = 'royalblue-40',
  Royalblue50 = 'royalblue-50',
  Royalblue60 = 'royalblue-60',
  Royalblue70 = 'royalblue-70',
  Royalblue80 = 'royalblue-80',
  Royalblue90 = 'royalblue-90',
  Royalblue100 = 'royalblue-100',

  Violet10 = 'violet-10',
  Violet20 = 'violet-20',
  Violet30 = 'violet-30',
  Violet40 = 'violet-40',
  Violet50 = 'violet-50',
  Violet60 = 'violet-60',
  Violet70 = 'violet-70',
  Violet80 = 'violet-80',
  Violet90 = 'violet-90',
  Violet100 = 'violet-100',

  Darkteal10 = 'darkteal-10',
  Darkteal20 = 'darkteal-20',
  Darkteal30 = 'darkteal-30',
  Darkteal40 = 'darkteal-40',
  Darkteal50 = 'darkteal-50',
  Darkteal60 = 'darkteal-60',
  Darkteal70 = 'darkteal-70',
  Darkteal80 = 'darkteal-80',
  Darkteal90 = 'darkteal-90',
  Darkteal100 = 'darkteal-100',

  Gold10 = 'gold-10',
  Gold20 = 'gold-20',
  Gold30 = 'gold-30',
  Gold40 = 'gold-40',
  Gold50 = 'gold-50',
  Gold60 = 'gold-60',
  Gold70 = 'gold-70',
  Gold80 = 'gold-80',
  Gold90 = 'gold-90',
  Gold100 = 'gold-100',
}

export enum DivergentColors {
  SkyTeal = 'skyblue-darkteal',
  VioletRoyal = 'violet-royalblue',
  VioletTeal = 'violet-darkteal',
  GoldTeal = 'gold-darkteal',
  GoldRoyal = 'gold-royalblue',
}

export enum OtherColors {
  Grey = 'grey',
  Green = 'green',
  Orange = 'orange',
  Red = 'red',
}

export enum LegacyColors {
  '01' = '01',
  '02' = '02',
  '03' = '03',
  '04' = '04',
  '05' = '05',
  '06' = '06',
  '07' = '07',
  '08' = '08',
  '09' = '09',
}

export type Color = Colors | DivergentColors | LegacyColors;

export type Margins = {
  top: number;
  right: number;
  bottom: number;
  left: number;
};

export type Orientation = 'horizontal' | 'vertical';

export const ORIENTATIONS: Record<string, Orientation> = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export type BarType = 'single' | 'grouped' | 'stacked';

export const BAR_TYPES: Record<string, BarType> = {
  SINGLE: 'single',
  GROUPED: 'grouped',
  STACKED: 'stacked',
};

export const DEFAULT_MARGINS = {
  VERTICAL: {
    top: 16,
    right: 0,
    bottom: 28,
    left: 24,
  } as Margins,
  HORIZONTAL: {
    top: 16,
    right: 0,
    bottom: 28,
    left: 24,
  } as Margins,
};

export const NO_DATA = 'No data';

export const BAR_HEIGHT = 20;

export const PADDING_VERTICAL = 0.33; // space between bars is 1/2 of a bar's width
export const PADDING_HORIZONTAL = 0.5; // space between bars is a bar's width

// Default radius for the tooltip anchor circle.
// Adding a negligible radius for tooltip anchor, as firefox doesn't respect tooltip element positioning without the circle having actual radius.
export const TOOLTIP_ANCHOR_RADIUS = 1;

export const TOOLTIP_POSITIONS = [
  'auto',
  'auto-start',
  'auto-end',
  'top',
  'top-start',
  'top-end',
  'right',
  'right-start',
  'right-end',
  'bottom',
  'bottom-start',
  'bottom-end',
  'left',
  'left-start',
  'left-end',
] as const; // so that type is not string[]
