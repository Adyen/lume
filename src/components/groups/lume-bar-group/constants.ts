export const DEFAULT_COLOR = 'skyblue';
