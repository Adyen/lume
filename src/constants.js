export const ORIENTATIONS = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export const DEFAULT_MARGINS = {
  VERTICAL: {
    top: 44,
    right: 0,
    bottom: 32,
    left: 32,
  },
  HORIZONTAL: {
    top: 44,
    right: 0,
    bottom: 32,
    left: 32,
  },
};

export const NO_DATA = 'No data';
