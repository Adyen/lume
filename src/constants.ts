type Margins = {
  top: number;
  right: number;
  bottom: number;
  left: number;
};

export type Orientation = 'horizontal' | 'vertical';

export const ORIENTATIONS: Record<string, Orientation> = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export type BarType = 'single' | 'grouped' | 'stacked';

export const BAR_TYPES: Record<string, BarType> = {
  SINGLE: 'single',
  GROUPED: 'grouped',
  STACKED: 'stacked',
};

export const DEFAULT_MARGINS = {
  VERTICAL: {
    top: 44,
    right: 0,
    bottom: 32,
    left: 32,
  } as Margins,
  HORIZONTAL: {
    top: 44,
    right: 0,
    bottom: 32,
    left: 32,
  } as Margins,
};

export const NO_DATA = 'No data';
