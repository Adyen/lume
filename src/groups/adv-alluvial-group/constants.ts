import { sankeyJustify } from 'd3-sankey';

export const BASE_DATA = {
  values: [],
  nodePadding: 0,
  nodeWidth: 15,
  nodeAlign: sankeyJustify,
  valueFormatter: (value: number) => String(value),
};

export const NODE_LABEL_PADDING = 8;
