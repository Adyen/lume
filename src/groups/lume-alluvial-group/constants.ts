export const DEFAULT_COLOR = '03';
export const NODE_LABEL_PADDING = 8;
