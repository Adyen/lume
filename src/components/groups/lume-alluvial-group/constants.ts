import { Colors } from '@/constants';

export const DEFAULT_COLOR = Colors.Skyblue;
export const NODE_LABEL_PADDING = 8;

export const GHOST_STROKE_WIDTH_OFFSET = 8;
