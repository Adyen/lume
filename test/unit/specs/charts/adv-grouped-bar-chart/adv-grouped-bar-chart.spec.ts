import { mount } from '@vue/test-utils';
import { data, labels, xScale, yScale, generateData } from '../../mock-data';
import GroupedBarChart from '@/charts/adv-grouped-bar-chart/adv-grouped-bar-chart.vue';
import { Orientation } from '@/constants';

const orientation: Orientation = 'horizontal';
const numberOfBars = data[0].values.length;

describe('adv-grouped-bar-chart.vue', () => {
    test('mounts component and sets prop values', () => {
        const wrapper = mount(GroupedBarChart, {
            propsData: { data, labels, xScale, yScale }
        });

        const el = wrapper.find('[data-j-grouped-bar-chart]')
        expect(el.exists()).toBeTruthy();
        expect(el.find('[data-j-bars-group]').exists()).toBeTruthy();
        const barsGroupComponent = el.find('[data-j-bars-group]');
        expect(barsGroupComponent.props()['orientation']).toEqual('vertical');
        expect(el.findAll('[data-j-adv-bar]')).toHaveLength(numberOfBars);
    });

    test('mounts component and sets custom orientation', () => {
        const wrapper = mount(GroupedBarChart, {
            // Note that we need to flip the scales so as to feed band and linear scales correctly
            propsData: { data, labels, yScale: xScale, xScale: yScale, orientation }
        });

        const el = wrapper.find('[data-j-bars-group]');
        expect(el.props()['orientation']).toEqual('horizontal');
        expect(el.findAll('[data-j-adv-bar]')).toHaveLength(numberOfBars);
    });

    test('mounts component with double dataset', () => {
        const numberOfSets = 2;
        const manipulatedData = generateData(numberOfSets, data[0].values.length)
        const wrapper = mount(GroupedBarChart, {
            // Note that we need to flip the scales so as to feed band and linear scales correctly
            propsData: { data: manipulatedData, labels, yScale: xScale, xScale: yScale, orientation }
        });

        const el = wrapper.find('[data-j-bars-group]');
        expect(el.findAll('[data-j-adv-bar]')).toHaveLength(numberOfSets * numberOfBars);
    })
});
