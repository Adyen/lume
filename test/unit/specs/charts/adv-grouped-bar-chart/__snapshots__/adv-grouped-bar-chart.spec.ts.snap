// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="adv-chart-container adv-chart-container--minimum"
  data-j-adv-chart=""
  data-j-chart-container=""
  data-j-grouped-bar-chart=""
>
  <div
    class="adv-chart__header"
  >
    <!---->
     
    <div
      class="adv-chart__controls"
    />
     
    <!---->
     
    <div
      class="adv-chart-legend adv-chart__legend"
      data-j-chart-legend=""
    >
      <div
        class="adv-chart-legend__item"
        tabindex="0"
      >
        <span
          class="adv-chart-legend__circle adv-background-color--01"
          data-j-chart-legend__circle=""
        />
        
    
  
      </div>
    </div>
  </div>
   
  <svg
    class="adv-chart-container__svg"
    data-j-chart-container__root=""
    height="44"
  >
    <g
      class="adv-chart-container__group"
      data-j-chart-container__group=""
      transform="translate(24, 16)"
    >
       
      <!---->
       
      <g
        class="axis"
        data-j-adv-chart__x-axis=""
        data-j-axis=""
        transform="translate(0, 0)"
      >
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        0
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(91.42857142857143, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        1
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(182.85714285714286, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        2
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(274.2857142857143, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        3
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(365.7142857142857, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        4
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(457.14285714285717, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        5
      
            </text>
          </g>
           
          <!---->
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(548.5714285714286, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
              height="28"
              width="91.42857142857143"
            />
             
            <text
              class="axis__label"
              dy="0.8em"
              text-anchor="middle"
              x="45.714285714285715"
              y="8"
            >
              
        6
      
            </text>
          </g>
           
          <!---->
        </g>
      </g>
       
      <g
        class="axis"
        data-j-adv-chart__y-axis=""
        data-j-axis=""
        transform="translate(0, 0)"
      >
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 0)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        0
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 40)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        5
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 80)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        10
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 120)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        15
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 160)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        20
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 200)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        25
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 240)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        30
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 280)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        35
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 320)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        40
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 360)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        45
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 400)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        50
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 440)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        55
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
        <g
          class="axis__tick"
          data-j-axis__tick=""
          transform="translate(0, 480)"
        >
          <g
            class="axis__tick-label"
            data-j-axis__tick-label=""
            pointer-events="all"
          >
            <rect
              class="axis__ghost"
            />
             
            <text
              class="axis__label"
              dy="0.4em"
              text-anchor="end"
              x="-8"
            >
              
        60
      
            </text>
          </g>
           
          <line
            class="axis__grid-line"
            x1="0"
          />
        </g>
      </g>
       
      <g
        class="adv-bar-group"
        containerSize="[object Object]"
        data-j-bars-group=""
        labels="Monday,Tuesday,Wednesday,Thursday,Friday,Saturday,Sunday"
      />
       
      <g
        class="adv-overlay-group"
        data-j-adv-chart__overlay-group=""
      />
       
      <g />
    </g>
  </svg>
    
</div>
`;
